visualization/app/codeCharta/ui/codeMap/threeViewer/threeSceneService.ts

```diff
@@ -191,18 +190,55 @@ export class ThreeSceneService implements CodeMapPreRenderServiceSubscriber, Map
 
 			hoveredLabel.position.add(this.normedTransformVector)
 
+			this.toggleLineAnimation(false, hoveredLabel)
+
 			this.highlightedLabel = hoveredLabel
 		}
 	}
 
+	resetLineHighlight() {
+		this.highlightedLineIndex = -1
+		this.highlightedLine = null
+	}
+
 	resetLabel() {
 		if (this.highlightedLabel !== null) {
 			this.highlightedLabel.position.sub(this.normedTransformVector)
 			this.highlightedLabel.material.opacity = this.mapLabelColors.alpha
+
+			if (this.highlightedLineIndex > -1) {
+				this.toggleLineAnimation(true)
+			}
+
 			this.highlightedLabel = null
 		}
 	}
 
+	getHoveredLabelLineIndex(labels: Object3D[], label: any) {
+		const index = labels
+			.map(function (labelObject) {
+				return labelObject.uuid
+			})
+			.indexOf(label.uuid)
+
+		if (index >= 0) {
+			return index + 1
+		}
+	}
+
+	toggleLineAnimation(reset: boolean, hoveredLabel?: Object3D) {
+		const geometry = new Geometry()
+		const endPoint = reset
+			? new Vector3(this.highlightedLabel.position.x, this.highlightedLabel.position.y, this.highlightedLabel.position.z)
+			: new Vector3(hoveredLabel.position.x, hoveredLabel.position.y, hoveredLabel.position.z)
+
+		geometry.vertices.push(this.highlightedLine.geometry.vertices[0], endPoint)
+
+		const newLineForHighlightedLabel = new Line(geometry, this.highlightedLine.material)
+
+		this.labels.children.splice(this.highlightedLineIndex, 1, newLineForHighlightedLabel)
```

Comment:
I might be wrong but it seems that `this.highlihtedLineIndex` is not really needed as variable. It could instead just be an inlinded `getHoveredLabelLineIndex()` function as that's triggered anyway?

Reply:
```suggestion
		this.labels.children[this.highlightedLineIndex] = newLineForHighlightedLabel
```